File: Data Structure/Heap/LET_no1167_Minimum Cost to Connect Sticks.cpp
class Solution {
public:
    int connectSticks(vector<int>& sticks) {
        int res = 0;
        priority_queue<int, vector<int>, greater<int>> min_heap(sticks.begin(), sticks.end());
        
        while (min_heap.size() > 1) {
            int a = min_heap.top();
            min_heap.pop();
            int b = min_heap.top();
            min_heap.pop();
            res += a + b;
            min_heap.push(a + b);
        }

        return res;
    }
};

File: Data Structure/Heap/LET_no1167_Minimum Cost to Connect Sticks.py
class Solution:
    def connectSticks(self, sticks: List[int]) -> int:
        heapq.heapify(sticks)
        res = 0
        while len(sticks) > 1:
            a = heapq.heappop(sticks)
            b = heapq.heappop(sticks)
            cur = a + b
            res += cur
            heapq.heappush(sticks, cur)
        
        return res